File: dataset/human/python/file_9274.py
import streams, sequtils, strutils, algorithm, math, future, sets, tables, hashes
proc `ceilDiv`*[T](x, y: T): T = x div y + ord(x mod y != 0)
proc `//=`*(x: var SomeInteger; y: SomeInteger) = x = x div y
proc `%=`*(x: var SomeInteger; y: SomeInteger) = x = x mod y

var
    L = stdin.readLine.split.map(parseInt)
    N = L[0]
    K = L[1]
    A = stdin.readLine.split.map(parseInt)
    route = newSeq[int]()
    state = 0
    visited = newSeqWith(N, false)
visited[0] = true
route.add state
while true:
    state = A[state] - 1
    route.add state
    if visited[state]:
        break
    visited[state] = true

if route.len > K:
    echo route[K] + 1
else:
    var
        idx = 0
        cnts = newSeqWith(N, 0)
        ans:int
    for i in 0..route.len-1:
        cnts[route[i]] += 1
    for i in 0..N-1:
        if cnts[i] == 2:
            idx = route.find(i)
            break
    K -= idx
    ans = route[idx..^2][K mod route[idx..^2].len] + 1
    echo ans

File: dataset/human/java/file_9274.java
import java.util.*;
public class Main {
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		int a = sc.nextInt();
		int b = sc.nextInt();
        long [] ans = new long [a+1];
        if(b>0){int [] in = new int [b];
        for(int i=0;i<b;i++){
                             in[i] = sc.nextInt();
                            }
		int c = 0;
        long mod = 1000000007;
        ans [0] = 1;
        if(in[c]==1){c++;}
        else{ans[1]=1;}
        for(int i=2;i<=a;i++){if(c<b && in[c]==i){c++;}
                              else{ans[i]=(ans[i-1]+ans[i-2])%mod;}
                             }       
		System.out.println(ans[a]);}
        else{long mod = 1000000007;
        ans [0] = 1;
        ans [1] = 1;
        for(int i=2;i<=a;i++){
                              ans[i]=(ans[i-1]+ans[i-2])%mod;
                             }       
		System.out.println(ans[a]);}
	}
}
